game/dino_jump_main.py

```diff
@@ -0,0 +1,394 @@
+                self.COOLDOWN_VARIANCE
+            )
+
+        # Otherwise, we decrement the cooldown timer every loop
+        else:
+            self.curr_cooldown -= 1
+
+
+#############################################################################
+# DINO MODULE
+#############################################################################
+
+
+class Dino:
+
+    def __init__(self, GROUND_AXIS):
+
+        # Jumping Constants
+        self.JUMP_HEIGHT = 25  # specify how many pixels upwards the max-height goes
+        self.JUMP_TIME = 500  # specify time in milliseconds dino takes to land
+        self.UPDATE_WALK_FRAMES = 3  # how many frames to wait before dino switches legs
+
+        # Inherited from Game
+        self.GROUND_AXIS = GROUND_AXIS
+
+        # Get all pixel arrays for the dino from sprite import (dino:20x19)
+        self.JUMP_ARRAY = import_sprite("jump")
+        self.RRUN_ARRAY = import_sprite("right run")
+        self.LRUN_ARRAY = import_sprite("left run")
+        self.RDUCK_ARRAY = import_sprite("right duck")
+        self.LDUCK_ARRAY = import_sprite("left duck")
+        self.DONE_ARRAY = import_sprite("done")
+
+        # Define Array and Dimensions
+        self.ARRAY = self.JUMP_ARRAY  # the first array is still dino frame
+        self.HEIGHT, self.WIDTH = (
+            self.JUMP_ARRAY.shape
+        )  # get height and widrh to be able to call
+        self.OG_DINO_Y = (
+            self.GROUND_AXIS - self.HEIGHT
+        )  # define dino's top left pixel height
+        self.x = 5  # dino x stays put, how many pixels from left edge
+        self.y = self.OG_DINO_Y  # assigned to the y position
+        self.is_right_foot_down = (
+            True  # used to change between left and right foot arrays displayed
+        )
+        self.is_jumping = True  # detects if jumping signal is given
+        self.is_ducking = False  # detects if ducking signal is given
+        self.is_hit = False  # becomes true when obstacle and dino collide
+        self.jump_start_time = 0  # used to count time in jumping
+        self.curr_walk = (
+            self.UPDATE_WALK_FRAMES
+        )  # counts how many frames before the legs switch
+
+        # Calculates Time Constants (physically realistic jumping)
+        self.TIME_TO_PEAK = self.JUMP_TIME / 2
+        self.ACCEL = (2 * self.JUMP_HEIGHT) / (self.TIME_TO_PEAK**2)
+        self.INIT_V = self.TIME_TO_PEAK * self.ACCEL
+        self.JUMP_COEF = self.JUMP_HEIGHT / (
+            self.TIME_TO_PEAK**2 - self.JUMP_TIME * self.TIME_TO_PEAK
+        )
+
+    def update_jumping_location(self):
+        # Using timing to determine if we are mid jump and y position should be adjusted
+        if pygame.time.get_ticks() - self.jump_start_time <= self.JUMP_TIME:
+            jump_time = pygame.time.get_ticks() - self.jump_start_time
+            self.y = self.OG_DINO_Y - self.JUMP_COEF * (
+                jump_time**2 - self.JUMP_TIME * jump_time
+            )

```

Comment:
I still think `JUMP_TIME` might be a bit confusing and `JUMP_DURATION` might be more clear it is less a time/instant and more a period of time for jumping.
Also as mentioned last time, a bit concerned about the difference treatment of time here, where we use actual "time" when the rest of the time uses "frames". But it's your call whether we care about fixing that now.

Reply:
I implemented and changed it to a frame based counting system now.